plugin/register.go

```diff
@@ -0,0 +1,57 @@
+package plugin
+
+import (
+	"fmt"
+	"github.com/mitchellh/cli"
+	"reflect"
+	"strings"
+	"trellis-cli/cmd"
+)
+
+func Register(c *cli.CLI, searchPaths []string, validPluginFilenamePrefixes []string) {
+	coreNamespaces := namespacesFor(reflect.ValueOf(c.Commands))
+	pluginFinder := newFinder(validPluginFilenamePrefixes, searchPaths, coreNamespaces)
+	plugins := pluginFinder.find()
+	pluginNamespaces := namespacesFor(reflect.ValueOf(plugins))
+
+	// Register plugin commands.
+	for name, bin := range plugins {
+		pluginCommand := cmd.NewPassthroughCommand(name, bin, c.Args)
+		c.Commands[name] = func() (cli.Command, error) {
+			return pluginCommand, nil
+		}
+	}
+
+	// Separate plugin commands from core command lists
+	c.HiddenCommands = append(c.HiddenCommands, pluginNamespaces...)
+	// Append plugin command list to help text.
+	c.HelpFunc = helpFunc(c.Name, pluginNamespaces)
+}
+
+func namespacesFor(v reflect.Value) []string {
```

Comment:
You can use a named return value parameter here if you want to avoid the `var` later. Personally I think it's a little clearer that the var being declared is the return value.

```go
func namespacesFor(v reflect.Value) (namespaces []string) {
}
```

note: this won't work with maps since their "zero" value is `nil` and not an empty map

Reply:
done in [`c241027` (#144)](https://github.com/roots/trellis-cli/pull/144/commits/c241027becb2641840f9245a88baa589cc235925)